### ISSUE TITLE
[bitnami/jenkins] cannot upgrade to the latest version (9.0.2) 

### ISSUE THREAD

**Issue by Stark-X** [bitnami/charts#9221]
### Name and Version

bitnami/jenkins 9.0.2

### What steps will reproduce the bug?

``` bash
$ export JENKINS_PASSWORD=$(kubectl get secret --namespace devops-jenkins ci-jenkins -o jsonpath="{.data.jenkins-password}" | base64 --decode)
$ helm upgrade -f ci/jenkins/values.yml ci-jenkins bitnami/jenkins -n devops-jenkins --set jenkinsPassword=$JENKINS_PASSWORD
```

Output:
```
Error: UPGRADE FAILED: template: jenkins/templates/metrics-svc.yaml:13:18: executing "jenkins/templates/metrics-svc.yaml" at <include "common.tplvalues.render" (dict "value" .Values.metrics.service.annotations "context" $)>: error calling include: template: jenkins/charts/common/templates/_tplvalues.tpl:11:12: executing "common.tplvalues.render" at <tpl (.value | toYaml) .context>: error calling tpl: error during tpl function execution for "prometheus.io/port: '{{ .Values.metrics.service.ports.http }}'\nprometheus.io/scrape: \"true\"": template: jenkins/templates/metrics-svc.yaml:1:31: executing "jenkins/templates/metrics-svc.yaml" at <.Values.metrics.service.ports.http>: nil pointer evaluating interface {}.http
```

### Are you using any custom parameters or values?

``` yaml
# helm install -f values.yaml ci-jenkins bitnami/jenkins -n devops-jenkins
image:
  tag: 2.319.3-debian-10-r1
jenkinsUser: devops
# provide to web display
jenkinsHost: *****
# extraEnvVars:
  # - file.encoding: "UTF-8"
  # - sun.jnu.encoding: "UTF-8"
  # - LANG: "en_US.UTF-8"
hostAliases: []
ingress:
  enabled: true
  # hostname that ingress generated
  hostname: *****
  extraHosts:
    - name: "******"
  annotations:
    nginx.ingress.kubernetes.io/proxy-body-size: '50m'
resources:
  limits:
    cpu: 2000m
    memory: 8Gi
  requests:
    cpu: 500m
    memory: 1Gi

persistence:
  size: 100Gi
  storageClass: nfs-client
  mode:
    - ReadWriteMany
metrics:
  enabled: true
service:
  type: ClusterIP
extraEnvVars:
  - name: LANG
    value: en_US.UTF-8
javaOpts:
  - "-Dfile.encoding=UTF-8"
  - "-Dsun.jnu.encoding=UTF-8"
  - "-Dhudson.slaves.NodeProvisioner.initialDelay=5"
  - "-Dhudson.slaves.NodeProvisioner.MARGIN=50"
  - "-Dhudson.slaves.NodeProvisioner.MARGIN0=0.85"
  - "-Duser.timezone=Asia/Shanghai"
  - "-Xmx8192m"
  - "-XshowSettings:vm"
# support on Chart version 8.20, before that version, just edit the deployment to make it effect
# you should create the ServiceAccount ci-jenkins first
serviceAccountName: ci-jenkins
```

### What is the expected behavior?

Upgrade correctly

### What do you see instead?

```
Error: UPGRADE FAILED: template: jenkins/templates/metrics-svc.yaml:13:18: executing "jenkins/templates/metrics-svc.yaml" at <include "common.tplvalues.render" (dict "value" .Values.metrics.service.annotations "context" $)>: error calling include: template: jenkins/charts/common/templates/_tplvalues.tpl:11:12: executing "common.tplvalues.render" at <tpl (.value | toYaml) .context>: error calling tpl: error during tpl function execution for "prometheus.io/port: '{{ .Values.metrics.service.ports.http }}'\nprometheus.io/scrape: \"true\"": template: jenkins/templates/metrics-svc.yaml:1:31: executing "jenkins/templates/metrics-svc.yaml" at <.Values.metrics.service.ports.http>: nil pointer evaluating interface {}.http
```

**Comment by javsalgar** [comment]
Hi,

Could you let us know from which version of the chart you are trying to upgrade?

**Comment by Stark-X** [comment]
> Hi,
> 
> Could you let us know from which version of the chart you are trying to upgrade?

jenkins-8.1.5

**Comment by javsalgar** [comment]
Hi,

In principle, the standardizations in the values changed the format to `metrics.service.ports.http`. It seems that the values.yaml was not correctly changed. Would you like to submit a PR fixing the issue?

**Comment by Stark-X** [comment]
Closes by #9368
